File: src/main/java/com/gmail/borlandlp/minigamesdtools/arena/localevent/ArenaPlayerDamagedLocalEvent.java
package com.gmail.borlandlp.minigamesdtools.arena.localevent;

import org.bukkit.entity.Player;
import org.bukkit.event.Cancellable;

public class ArenaPlayerDamagedLocalEvent extends ArenaEvent implements Cancellable {
    private Player player;
    private double damage;
    private boolean canceled;

    public ArenaPlayerDamagedLocalEvent(Player p, double d) {
        this.player = p;
        this.damage = d;
    }

    public Player getPlayer() {
        return player;
    }

    public double getDamage() {
        return damage;
    }

    @Override
    public boolean isCancelled() {
        return this.canceled;
    }

    @Override
    public void setCancelled(boolean b) {
        this.canceled = b;
    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/arena/localevent/ArenaPlayerDamagedLocalEvent.kt
package com.gmail.borlandlp.minigamesdtools.arena.localevent

import org.bukkit.entity.Player
import org.bukkit.event.Cancellable

class ArenaPlayerDamagedLocalEvent(val player: Player, val damage: Double) : ArenaEvent(),
    Cancellable {
    private var canceled = false

    override fun isCancelled(): Boolean {
        return canceled
    }

    override fun setCancelled(b: Boolean) {
        canceled = b
    }

}